File: Ex3/Task3.6/FishTankMaterials.scala
/**
 * Takes the dimensions of a fish tank as command
 * line arguments and prints the surface area and
 * the total length of the edges
 */
object FishTankMaterials {
    def main(args: Array[String]) {
        var width: Int = args(0).toInt
        var depth: Int = args(1).toInt
        var height: Int = args(2).toInt

        var surfaceArea: Int = 2 * (width * height + width * depth + height * depth)

        var edgesLength: Int = 4 * (height + depth + width)

        println("The surface area of a tank with dimensions ("
            + width + ", " + depth + ", " + height + ") is " + surfaceArea)

        println("The length of the edges of a tank with dimensions ("
            + width + ", " + depth + ", " + height + ") is " + edgesLength)
    }
}

File: ex3/task3.6/FishTankMaterials.java
/**
* Prints the surface area of a tank and the lenght of the edges
* Contained calculus errors
* Errors fixed
* For values (3.2, 2.5, 0)/(1.2, 2.2, 3.1) program crashed 
* For values (10, 20, 30) the result was the exepcted one
* Also tested with negative numbers, inappropiate result
* @author Vlad Georgescu 
*/
public class FishTankMaterials
{
    public static void main(String[] args)
    {
        int width = Integer.parseInt(args[0]);
        int depth = Integer.parseInt(args[1]);
        int height = Integer.parseInt(args[2]);

        int surfaceArea = 2 * ( width * height + width * depth + height * depth);
        
        int edgesLenght = 4 * ( width + depth + height );

        System.out.println("The surface area of a tank with dimensions " 
                           + "(" + width + "," + depth + "," + height + ") "  
                           + "is " + surfaceArea);

        System.out.println("The lenght of the edges of a tank with dimensions "
                           + "(" + width + "," + depth + "," + height + ") "
                           + "is " + edgesLenght);                   

    }
}